src/Microsoft.MobileBlazorBindings.Authentication/Http/ApiAuthorizationMessageHandler.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Net.Http;
+
+namespace Microsoft.MobileBlazorBindings.Authentication
+{
+    /// <summary>
+    /// A <see cref="DelegatingHandler"/> that attaches access tokens to outgoing <see cref="HttpRequestMessage"/> instances.
+    /// Access tokens will only be added when the request URI is within the API's base URI.
+    /// </summary>
+    /// <remarks>
+    /// Since message handlers have their own scope to resolve from, <see cref="RegisterTokenProvider(string, IAccessTokenProvider)"/>
+    /// needs to be called to register the correct <see cref="IAccessTokenProvider"/> for the provided base URL.
+    /// </remarks>
+    public class ApiAuthorizationMessageHandler : AuthorizationMessageHandler
```

Comment:
I don't really get why do you need ApiAuthorizationMessageHandler separately of AuthorizationMessageHandler. What's the point?

Reply:
Not much, except that Blazor WebAssembly has more or less the same construct (only there it's called BaseAddressAuthorizationMessageHandler). I tried to keep stuff like message handlers and service registration mostly compatible with WebAssembly auth so that we can either merge/share codebases in the future and provide familiarity.